### PR TITLE
feat: add Magic bridge queries and wiring

### PR DESCRIPTION
**Pull request by ldiego08**
# Changes

- Add React hooks and fetch functions to interface with the Magic contracts and the Bitcoin network.
- Add typings for the Magic contracts the Clarigen client can use to access functions strongly-typed.
- Add packages and dependencies (microstacks, Clarigen, etc.)

> Try out this version of the Leather — download [extension builds](https://github.com/leather-wallet/extension/actions/runs/6086690172).

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205346192856258

### CODE REVIEW

File: src/app/common/magic/contracts.ts
Comment (ldiego08) [inline src/app/common/magic/contracts.ts:1]: These types were generated from the on-chain Magic contracts using Clarigen. In order not to add Clarigen and the contracts to the wallet repo (+20 files), we can keep this file here as is.

File: src/app/common/electrum/client.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        port: 443,
+      };
+
+    default:
+      return {
+        protocol: 'ssl',
+        host: 'blackie.c3-soft.com',
+        port: 57006,
+      };
+  }
```
Comment (ldiego08) [inline src/app/common/electrum/client.ts:27]: @kyranjamie just want to make sure it's fine to use these 3rd party services for Electrum.

File: src/app/common/magic/utils.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
-import { base58checkEncode, hashRipemd160 } from 'micro-stacks/crypto';
-import { hashSha256 } from 'micro-stacks/crypto-sha';
-
 import { DefaultNetworkConfigurations } from '@shared/constants';
 
 import { getBitcoinNetwork } from './network';
+import { sha256 } from '@noble/hashes/sha256';
+import { ripemd160 } from '@noble/hashes/ripemd160';
+import { toBase58Check } from 'bitcoinjs-lib/src/address';
```
Comment (kyranjamie) [inline src/app/common/magic/utils.ts]: Also in scure-base package https://github.com/paulmillr/scure-base#usage

File: src/app/common/magic/utils.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { ripemd160 } from '@noble/hashes/ripemd160';
+import { sha256 } from '@noble/hashes/sha256';
+import { toBase58Check } from 'bitcoinjs-lib/src/address';
+
+import { DefaultNetworkConfigurations } from '@shared/constants';
+
+import { getBitcoinNetwork } from './network';
+
+export function pubKeyToBtcAddress(publicKey: Uint8Array, network: DefaultNetworkConfigurations) {
```
Comment (fbwoolf) [inline src/app/common/magic/utils.ts:9]: Maybe this would be better located in `bitcoin.utils.ts`?

**Comment by fbwoolf** [comment]
@kyranjamie @markmhx should we get this merged soon so it doesn't get too outdated?

**Comment by fbwoolf** [comment]
Closing this in favor of moving the code to an sdk.

**Pull request closed without merging**